.gitlab-ci.yml

```diff
@@ -0,0 +1,70 @@
+include:
+  - project: hpc/gitlab-pipelines
+    ref: olupton/coreneuron # TODO drop this
+    file: spack-build-components.gitlab-ci.yml
+
+# Configure when the pipeline actually runs.
+workflow:
+  rules:
+    # Don't run on push events for now to try and save resources
+    - if: '$CI_PIPELINE_SOURCE == "push"'
+      when: never
+    # Don't run on MR events because we also get "external_pull_request_event" from GitHub.
+    - if: '$CI_PIPELINE_SOURCE == "merge_request_event"'
+      when: never
+    - when: always
+
+stages:
+    - .pre
+    - build
+    - test
+    - build_neuron
+    - test_neuron
+
+# Set up Spack
+spack_setup:
+  extends: .spack_setup_ccache
+
+.spack_intel:
+  variables:
+    SPACK_COMPILER: intel
+.spack_coreneuron:
+  variables:
+    SPACK_PACKAGE: coreneuron
+    SPACK_SPEC: +debug+tests~legacy-unit
+.spack_neuron:
+  variables:
+    SPACK_PACKAGE: neuron
+    # TODO: need a way of configuring this, maybe from the PR description?
```

Comment:
We have use this approach already for Jenkins so same can be used?

Reply:
I think we can re-use some commands from Jenkins, but it will also need some new plumbing.

This can be addressed later. In the short term we can just modify `.gitlab-ci.yml` in open PRs that are waiting for changes to be merged into NEURON, as was done here.

Reply:
I opened #505 to track this.